File: ms-flight-search/src/main/java/dev/patricksilva/model/services/FlightSearchService.java
package dev.patricksilva.model.services;

import java.util.List;
import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import dev.patricksilva.model.repositories.FlightSearchRepository;

@Service
public class FlightSearchService {

	@Autowired
    private FlightSearchRepository flightSearchRepository;

    public List<Map<String, Object>> searchFlights(String cidadeOrigem, String cidadeDestino, String partidaPrevista) {
    	return flightSearchRepository.searchFlights(cidadeOrigem, cidadeDestino, partidaPrevista);
    }
}

File: ms-GO-flight-search/model/services/FlightSearchService.go
package services

import (
	"ms-GO-flight-search/model/repositories"
)

type FlightSearchService struct {
	FlightSearchRepository repositories.FlightSearchRepository
}

func NewFlightSearchService(repo repositories.FlightSearchRepository) *FlightSearchService {
	return &FlightSearchService{
		FlightSearchRepository: repo,
	}
}

func (service *FlightSearchService) SearchFlights(cidadeOrigem, cidadeDestino, partidaPrevista string) ([]map[string]interface{}, error) {
	return service.FlightSearchRepository.SearchFlights(cidadeOrigem, cidadeDestino, partidaPrevista)
}
